pkg/challenge/challenger.go

```diff
@@ -18,16 +19,105 @@ type OuterChallenge struct {
 	Msg []byte `msgpack:"msg"`
 }
 
+func (o *OuterChallenge) Verify(counterParty ecdsa.PublicKey) error {
+	return echelper.VerifySignature(counterParty, o.Msg, o.Sig)
+}
+
+func (o *OuterChallenge) RequestData() ([]byte, error) {
+	inner, err := o.inner()
+	if err != nil {
+		return nil, err
+	}
+	return inner.RequestData, nil
+}
```

Comment:
I think move the `inner()` call to inside Verify will make callers clean

Reply:
done